### PR TITLE
Feat div divw

### PR DESCRIPTION
**Pull request by smml1996**
adds support for div and divw instructions. Only adds a node for DIV and DIVW is built using DIV

### CODE REVIEW

File: src/unicorn/optimize.rs [hunk 1]
```diff
@@ -569,6 +569,11 @@ impl<'a, S: SMTSolver> ConstantFolder<'a, S> {
                 if let Some(n) = self.visit(right) { *right = n }
                 self.fold_mul(left, right)
             }
+            Node::Divu { ref mut left, ref mut right, .. } => {
+                if let Some(n) = self.visit(left) { *left = n }
+                if let Some(n) = self.visit(right) { *right = n }
+                self.fold_div(left, right)
```
Comment (mstarzinger) [inline src/unicorn/optimize.rs]: The current `fold_div` is based on `u64::checked_div`, I don't think we can reuse that for the signed `Node::Div` below, I think we need `fold_div` and `fold_udiv` with different semantics.

**Review by mstarzinger — COMMENTED** [review]
One comment in optimize.rs, I haven't thoroughly reviewed the changes to bitblasting.rs though.

File: src/unicorn/optimize.rs (hunk 1 above)
Comment (smml1996) [reply src/unicorn/optimize.rs]: Yes, I forgot about that when I was refactoring. Thanks for catching the bug!

**Review by mstarzinger — APPROVED** [review]
Looks good. Thanks! I didn't review changes to bitblasting.rs in detail.

**Pull request merged**